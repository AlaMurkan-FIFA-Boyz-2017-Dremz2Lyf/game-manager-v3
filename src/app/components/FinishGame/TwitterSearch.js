import React from 'react';
import PropTypes from 'prop-types';
import { Segment, Message, Header, Icon } from 'semantic-ui-react';

const TwitterSearch = ({ hashtag }) => (
  <Segment basic>
    <Header color="teal" as="h2">
      <Icon name="twitter" />
      <Header.Content>
        Tweet it.
      </Header.Content>
    </Header>
    <Message color="teal">
      <Message.Header>
        {`#${hashtag}`}
      </Message.Header>
      <p>Add this hashtag to your tweet and next after the tweet has been sent.</p>
    </Message>
  </Segment>
);

TwitterSearch.propTypes = {
  hashtag: PropTypes.string,
  generateHashtag: PropTypes.func,
};

export default TwitterSearch;
