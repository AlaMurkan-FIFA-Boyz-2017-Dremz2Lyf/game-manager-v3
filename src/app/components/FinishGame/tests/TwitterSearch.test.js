import React from 'react';
import { shallow } from 'enzyme';

import TwitterSearch from '../TwitterSearch';

const hashtag = 'fifaboyz';

describe('TwitterSearch', () => {
  it('should match the snapshot and show the hashtag based on props', () => {
    const wrapper = shallow(
      <TwitterSearch hashtag={hashtag} />,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('MessageHeader').html()).toBe(`<div class="header">#${hashtag}</div>`);
  });
});
