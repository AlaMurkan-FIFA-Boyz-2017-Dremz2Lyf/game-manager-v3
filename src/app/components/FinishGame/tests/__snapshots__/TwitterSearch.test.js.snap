// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitterSearch should match the snapshot and show the hashtag based on props 1`] = `"<div class=\\"ui basic segment\\"><h2 class=\\"ui teal header\\"><i aria-hidden=\\"true\\" class=\\"twitter icon\\"></i><div class=\\"content\\">Tweet it.</div></h2><div class=\\"ui teal message\\"><div class=\\"header\\">#fifaboyz</div><p>Add this hashtag to your tweet and next after the tweet has been sent.</p></div></div>"`;
